packages/elements/src/datetime-field/index.ts

```diff
@@ -0,0 +1,742 @@
+import { Direction } from './constants.js';
+import {
+  NavigationKeys,
+  DateTimeFormatPart,
+  InputSelection
+} from './types';
+import { TextField } from '../text-field/index.js';
+import {
+  getSelectedPartIndex,
+  getNextSelectedPartIndex,
+  selectPart
+} from './utils.js';
+import { VERSION } from '../version.js';
+import '@refinitiv-ui/phrasebook/locale/en/datetime-field.js';
+
+/**
+ * A form control element for datetime input.
+ *
+ * @fires value-changed - Dispatched when value changes
+ * @fires error-changed - Dispatched when error state changes
+ * @fires icon-click - Dispatched when icon is clicked
+ *
+ * @attr {boolean} disabled - Set disabled state
+ * @prop {boolean} [disabled=false] - Set disabled state
+ *
+ * @attr {boolean} error - Set error state
+ * @prop {boolean} [error=false] - Set error state
+ *
+ * @attr {string} icon - Specify icon to display in input. Value can be icon name
+ * @prop {string | null} [icon=null] - Specify icon to display in input. Value can be icon name
+ *
+ * @attr {boolean} icon-has-action - Specify when icon need to be clickable
+ * @prop {boolean} [iconHasAction=false] - Specify when icon need to be clickable
+ *
+ * @attr {number} maxlength - Set character max limit
+ * @prop {number | null} [maxLength=null] - Set character max limit
+ *
+ * @attr {number} minlength - Set character min limit
+ * @prop {number | null} [minLength=null] - Set character min limit
+ *
+ * @prop {string} [pattern=""] - Set regular expression for input validation
+ *
+ * @attr {string} placeholder - Set placeholder text
+ * @prop {string} [placeholder=""] - Set placeholder text
+ *
+ * @attr {boolean} readonly - Set readonly state
+ * @prop {boolean} [readonly=false] - Set readonly state
+ *
+ * @attr {boolean} transparent - Disables all other states and border/background styles.
+ * @prop {boolean} [transparent=false] - Disables all other states and border/background styles.
+ *
+ * @attr {boolean} warning - Set warning state
+ * @prop {boolean} [warning=false] - Set warning state
+ */
+@customElement('ef-datetime-field')
+export class DatetimeField extends TextField {
+  /**
+   * Element version number
+   * @returns version number
+   */
+  static get version (): string {
+    return VERSION;
+  }
+
+  /**
+   * Delay selecting a part to ensure
+   * that scroll can be updated
+   */
+  private selectPartFrame = new AnimationTaskRunner();
+
+  /**
+   * Set minimum date.
+   * This value must follow the `format` and be less
+   * than or equal to the value of the `max` attribute
+   */
+  @property({ type: String, reflect: true })
+  public min: string | null = null;
+
+  /**
+   * Set maximum date.
+   * This value must follow the `format` and be greater
+   * than or equal to the value of the `min` attribute
+   */
+  @property({ type: String, reflect: true })
+  public max: string | null = null;
+
+  private _timepicker = false;
+  /**
+   * Toggle to display the time picker
```

Comment:
Should we also mention that, 
```suggestion
   * Toggle to display the time picker.
   * It is required in order to display time.
```
?

Reply:
I do not think so, the name of attribute is self descriptive without any comments